Update commitment options (#12173)
solana-labs/solana: docs/src/apps/jsonrpc-api.md

Solana nodes choose which bank state to query based on a commitment requirement
set by the client. Clients may specify either:

> - `"max"` - the node will query the most recent bank confirmed by the cluster as having reached `MAX_LOCKOUT_HISTORY` confirmations

Reviewer: ```suggestion
- `"max"` - the node will query the most recent bank confirmed by supermajority of the network as having reached `MAX_LOCKOUT_HISTORY` confirmations
```

Reviewer: (to be consistent with `single`)

- `"root"` - the node will query the most recent bank having reached `MAX_LOCKOUT_HISTORY` confirmations on this node

> - `"single"` - the node will query the most recent bank having reached 1 confirmation by supermajority of the network.

Reviewer: Re: use of "bank" here -- I feel we should use "block" instead as this information is more intended to be outward facing and we don't define "bank" at https://docs.solana.com/terminology

Solana nodes choose which bank state to query based on a commitment requirement
set by the client. Clients may specify either:

- `"max"` - the node will query the most recent block confirmed by supermajority of the network as having reached

> maximum lockout on this node.

Reviewer 2: ```suggestion
maximum lockout.
```

> - `"root"` - the node will query the most recent block having reached `MAX_LOCKOUT_HISTORY` confirmations on this node.

Reviewer 2: ```suggestion
- `"root"` - the node will query the most recent block having reached maximum lockout on this node.
```

Solana nodes choose which bank state to query based on a commitment requirement
set by the client. Clients may specify either:

- `"max"` - the node will query the most recent block confirmed by supermajority of the cluster as having reached
maximum lockout.
- `"root"` - the node will query the most recent block having reached maximum lockout on this node.
- `"singleGossip"` - the node will query the most recent block that has been voted on by supermajority of the cluster via

> gossip votes. This differs from `single` in that:

Reviewer: ```suggestion
gossip votes.
```
(single is no longer in here, so we can't reference it)

  - It incorporates votes from gossip and replay.

>   - It does not count votes on descendants of a bank, only direct votes on the bank.

Reviewer: ```suggestion
  - It does not count votes on descendants of a block, only direct votes on that block.
```

Unsubscribe from vote notifications

#### Parameters:

> s

Reviewer 2: Random "s"
